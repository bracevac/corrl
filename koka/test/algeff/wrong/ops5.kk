// ----------------------------------------------------------------------------
// operation are covered multiple times in a handler
// ----------------------------------------------------------------------------
 
effect state {
  get() : int; 
  set(i : int) : ();  
}

val state = handler(i:int) {
  return x -> x
  get()    -> resume(i,i)
  set(j)   -> resume(j,())
  get()    -> resume(0,i)
}
