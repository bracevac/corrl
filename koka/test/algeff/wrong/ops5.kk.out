test/algeff/wrong/ops5.kk(10,13): error: operator get is handled multiple times